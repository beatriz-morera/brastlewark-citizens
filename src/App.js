import React from 'react';
import { HashRouter, Switch, Link, Route, Redirect } from 'react-router-dom';

import Home from '../src/Containers/Home';
import Search from '../src/Containers/Search';
import Detail from '../src/Containers/Detail';

const App = () => {
  return (
    <HashRouter>
      <header className="navigation">
        <Link to="/home">
          <h1 className="page-title">BRASTLEWARK</h1>
        </Link>
        <ul className="navigation-items">
          <li>
            <Link to="/home" className="navigation-elements">
              HOME
            </Link>
          </li>
          <li>
            <Link to="/search" className="navigation-elements">
              SEARCH
            </Link>
          </li>
        </ul>
      </header>
      <Switch>
        <Route path="/home" component={Home} exact={true} />
        <Route path="/search" component={Search} exact={true} />
        <Route path="/search/:id" component={Detail} />
        <Route path="/" render={() => <Redirect to="/home" />} exact={true} />
      </Switch>
    </HashRouter>
  );
};

export default App;
