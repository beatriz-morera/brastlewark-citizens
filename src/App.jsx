import React from 'react';
import { HashRouter, Switch, Link, Route, Redirect } from 'react-router-dom';

import Home from './pages/Home';
import Search from './pages/Search';
import Detail from './pages/Detail';

import CitizensProvider from './components/CitizensProvider';

const App = () => (
  <CitizensProvider>
    <HashRouter>
      <header className="navigation" data-testid="header">
        <Link to="/home" data-testid="home-link">
          <h1 className="page-title">BRASTLEWARK</h1>
        </Link>
        <ul className="navigation-items">
          <li>
            <Link to="/home" className="navigation-elements">
              HOME
            </Link>
          </li>
          <li>
            <Link to="/search" className="navigation-elements" data-testid="search-link">
              SEARCH
            </Link>
          </li>
        </ul>
      </header>
      <Switch>
        <Route path="/home" component={Home} exact />
        <Route path="/search" component={Search} exact />
        <Route path="/search/:id" component={Detail} />
        <Route path="/" render={() => <Redirect to="/home" />} exact />
      </Switch>
    </HashRouter>
  </CitizensProvider>
);

export default App;
