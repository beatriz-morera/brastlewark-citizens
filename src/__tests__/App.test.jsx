import React from 'react';

import {
  render,
  waitForDomChange,
  cleanup,
  fireEvent,
  waitForElement,
} from '@testing-library/react';
import App from '../App';

jest.mock('../services/citizens');

describe('app', () => {
  afterEach(cleanup);
  it('should render', async () => {
    expect.assertions(1);
    const { asFragment } = render(<App />);
    await waitForDomChange();
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render the home page', async () => {
    expect.assertions(1);
    const { getByTestId } = render(<App />);

    const homePage = await waitForElement(() => getByTestId('home-page'));

    expect(homePage).toBeInTheDocument();
  });

  it('should navigate to search page', async () => {
    expect.assertions(1);
    const { getByTestId } = render(<App />);

    fireEvent.click(getByTestId('search-link'));

    const searchPage = await waitForElement(() => getByTestId('search-page'));

    expect(searchPage).toBeInTheDocument();
  });
});
