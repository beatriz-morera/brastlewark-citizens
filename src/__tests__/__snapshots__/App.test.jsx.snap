// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app should render 1`] = `
<DocumentFragment>
  <header
    class="navigation"
    data-testid="header"
  >
    <a
      data-testid="home-link"
      href="#/home"
    >
      <h1
        class="page-title"
      >
        BRASTLEWARK
      </h1>
    </a>
    <ul
      class="navigation-items"
    >
      <li>
        <a
          class="navigation-elements"
          href="#/home"
        >
          HOME
        </a>
      </li>
      <li>
        <a
          class="navigation-elements"
          data-testid="search-link"
          href="#/search"
        >
          SEARCH
        </a>
      </li>
    </ul>
  </header>
  <section
    class="all-cards-container"
    data-testid="home-page"
  >
    <div
      class="ant-list ant-list-split ant-list-grid ant-list-something-after-last-item"
      data-testid="citizens"
    >
      <div
        class="ant-spin-nested-loading"
      >
        <div
          class="ant-spin-container"
        >
          <div
            class="ant-row"
          >
            <div
              class="ant-col ant-col-xs-24 ant-col-sm-12 ant-col-md-8 ant-col-lg-8 ant-col-xl-6 ant-col-xxl-4"
            >
              <div
                class="ant-list-item"
              >
                <a
                  href="#/search/0"
                >
                  <div
                    class="card-container"
                  >
                    <div
                      class="card"
                    >
                      <div
                        class="image-container"
                      >
                        <img
                          alt="Tobus Quickwhistle"
                          class="image"
                          src="http://www.publicdomainpictures.net/pictures/10000/nahled/thinking-monkey-11282237747K8xB.jpg"
                        />
                      </div>
                      <section
                        class="details-container"
                      >
                        <h4
                          class="citizen-name"
                        >
                          Tobus Quickwhistle
                        </h4>
                        <ul>
                          <li
                            class="details"
                          >
                            <span
                              class="details-title"
                            >
                              age:
                            </span>
                             306
                          </li>
                          <li
                            class="details"
                          >
                            <span
                              class="details-title"
                            >
                              weight:
                            </span>
                             39.065952
                          </li>
                          <li
                            class="details"
                          >
                            <span
                              class="details-title"
                            >
                              height:
                            </span>
                             107.75835
                          </li>
                        </ul>
                      </section>
                    </div>
                  </div>
                </a>
              </div>
            </div>
            <div
              class="ant-col ant-col-xs-24 ant-col-sm-12 ant-col-md-8 ant-col-lg-8 ant-col-xl-6 ant-col-xxl-4"
            >
              <div
                class="ant-list-item"
              >
                <a
                  href="#/search/1"
                >
                  <div
                    class="card-container"
                  >
                    <div
                      class="card"
                    >
                      <div
                        class="image-container"
                      >
                        <img
                          alt="Fizkin Voidbuster"
                          class="image"
                          src="http://www.publicdomainpictures.net/pictures/120000/nahled/white-hen.jpg"
                        />
                      </div>
                      <section
                        class="details-container"
                      >
                        <h4
                          class="citizen-name"
                        >
                          Fizkin Voidbuster
                        </h4>
                        <ul>
                          <li
                            class="details"
                          >
                            <span
                              class="details-title"
                            >
                              age:
                            </span>
                             288
                          </li>
                          <li
                            class="details"
                          >
                            <span
                              class="details-title"
                            >
                              weight:
                            </span>
                             35.279167
                          </li>
                          <li
                            class="details"
                          >
                            <span
                              class="details-title"
                            >
                              height:
                            </span>
                             110.43628
                          </li>
                        </ul>
                      </section>
                    </div>
                  </div>
                </a>
              </div>
            </div>
            <div
              class="ant-col ant-col-xs-24 ant-col-sm-12 ant-col-md-8 ant-col-lg-8 ant-col-xl-6 ant-col-xxl-4"
            >
              <div
                class="ant-list-item"
              >
                <a
                  href="#/search/2"
                >
                  <div
                    class="card-container"
                  >
                    <div
                      class="card"
                    >
                      <div
                        class="image-container"
                      >
                        <img
                          alt="Malbin Chromerocket"
                          class="image"
                          src="http://www.publicdomainpictures.net/pictures/30000/nahled/maple-leaves-background.jpg"
                        />
                      </div>
                      <section
                        class="details-container"
                      >
                        <h4
                          class="citizen-name"
                        >
                          Malbin Chromerocket
                        </h4>
                        <ul>
                          <li
                            class="details"
                          >
                            <span
                              class="details-title"
                            >
                              age:
                            </span>
                             166
                          </li>
                          <li
                            class="details"
                          >
                            <span
                              class="details-title"
                            >
                              weight:
                            </span>
                             35.88665
                          </li>
                          <li
                            class="details"
                          >
                            <span
                              class="details-title"
                            >
                              height:
                            </span>
                             106.14395
                          </li>
                        </ul>
                      </section>
                    </div>
                  </div>
                </a>
              </div>
            </div>
            <div
              class="ant-col ant-col-xs-24 ant-col-sm-12 ant-col-md-8 ant-col-lg-8 ant-col-xl-6 ant-col-xxl-4"
            >
              <div
                class="ant-list-item"
              >
                <a
                  href="#/search/3"
                >
                  <div
                    class="card-container"
                  >
                    <div
                      class="card"
                    >
                      <div
                        class="image-container"
                      >
                        <img
                          alt="Midwig Gyroslicer"
                          class="image"
                          src="http://www.publicdomainpictures.net/pictures/10000/nahled/1-1275919724d1Oh.jpg"
                        />
                      </div>
                      <section
                        class="details-container"
                      >
                        <h4
                          class="citizen-name"
                        >
                          Midwig Gyroslicer
                        </h4>
                        <ul>
                          <li
                            class="details"
                          >
                            <span
                              class="details-title"
                            >
                              age:
                            </span>
                             240
                          </li>
                          <li
                            class="details"
                          >
                            <span
                              class="details-title"
                            >
                              weight:
                            </span>
                             40.97596
                          </li>
                          <li
                            class="details"
                          >
                            <span
                              class="details-title"
                            >
                              height:
                            </span>
                             127.88554
                          </li>
                        </ul>
                      </section>
                    </div>
                  </div>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-list-pagination"
      >
        <ul
          class="ant-pagination mini"
          unselectable="unselectable"
        >
          <li
            aria-disabled="true"
            class="ant-pagination-disabled ant-pagination-prev"
            title="Previous Page"
          >
            <a
              class="ant-pagination-item-link"
            >
              <i
                aria-label="icon: left"
                class="anticon anticon-left"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="left"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 0 0 0 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                  />
                </svg>
              </i>
            </a>
          </li>
          <li
            class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
            tabindex="0"
            title="1"
          >
            <a>
              1
            </a>
          </li>
          <li
            aria-disabled="true"
            class="ant-pagination-disabled ant-pagination-next"
            title="Next Page"
          >
            <a
              class="ant-pagination-item-link"
            >
              <i
                aria-label="icon: right"
                class="anticon anticon-right"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="right"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M765.7 486.8L314.9 134.7A7.97 7.97 0 0 0 302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 0 0 0-50.4z"
                  />
                </svg>
              </i>
            </a>
          </li>
          <li
            class="ant-pagination-options"
          >
            <div
              class="ant-select-sm ant-pagination-options-size-changer ant-select ant-select-enabled"
            >
              <div
                aria-autocomplete="list"
                aria-controls="test-uuid"
                aria-expanded="false"
                aria-haspopup="true"
                class="ant-select-selection
            ant-select-selection--single"
                role="combobox"
                tabindex="0"
              >
                <div
                  class="ant-select-selection__rendered"
                >
                  <div
                    class="ant-select-selection-selected-value"
                    style="display: block; opacity: 1;"
                    title="10 / page"
                  >
                    10 / page
                  </div>
                </div>
                <span
                  class="ant-select-arrow"
                  style="user-select: none;"
                  unselectable="on"
                >
                  <i
                    aria-label="icon: down"
                    class="anticon anticon-down ant-select-arrow-icon"
                  >
                    <svg
                      aria-hidden="true"
                      class=""
                      data-icon="down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </i>
                </span>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
